src/main/java/com/google/api/generator/gapic/composer/defaultvalue/DefaultValueComposer.java

```diff
@@ -557,4 +592,24 @@ static AnonymousClassExpr createAnonymousResourceNameClass(String fieldOrMessage
         .setMethods(Arrays.asList(getFieldValuesMapMethod, getFieldValueMethod))
         .build();
   }
+
+  private static String constructValueMatchingPattern(String fieldName, String pattern) {
+    if (pattern == null || pattern.isEmpty()) {
+      return fieldName + fieldName.hashCode();
+    }
+
+    final String suffix = "-" + (Math.abs((fieldName + pattern).hashCode()) % 10000);
+
+    String value = pattern;
+    value = value.replace("**", "*");
+
+    String prevTempl = null;
+    while (!value.equals(prevTempl)) {
+      prevTempl = value;
+      value = REPLACER_PATTERN.matcher(value).replaceFirst("$1$2$1" + suffix);
+    }
+
+    value = value.replace("*", fieldName + suffix);
+    return value;
```

Comment:
```suggestion
    return value.replace("*", fieldName + suffix);
```

Reply:
Thanks, I'm surprised that Intellij does not highlight stuff like this =)